habitat-lab/habitat/tasks/rearrange/rearrange_sim.py

```diff
@@ -593,19 +593,25 @@ def _add_objs(
         if should_add_objects:
             self._scene_obj_ids = []
 
+        # Get Object template manager
+        otm = self.get_object_template_manager()
+
         for i, (obj_handle, transform) in enumerate(ep_info.rigid_objs):
             t_start = time.time()
             if should_add_objects:
-                template = None
-                for obj_path in self._additional_object_paths:
-                    template = osp.join(obj_path, obj_handle)
-                    if osp.isfile(template):
-                        break
-                assert (
-                    template is not None
-                ), f"Could not find config file for object {obj_handle}"
-
-                ro = rom.add_object_by_template_handle(template)
+
+                # Get object path
+                object_template = otm.get_templates_by_handle_substring(obj_handle)
+
+                # Exit if template is invalid
+                if not object_template:
+                    raise ValueError(f"Template not found for object with handle {obj_handle}")
+
+                # Get object path
+                object_path = list(object_template.keys())[0]
```

Comment:
This would be a bit clearer like so :

```suggestion
                # Get object path
                object_template = otm.get_first_matching_template_by_handle(obj_handle)

                # Exit if template is invalid
                if not object_template:
                    raise ValueError(f"Template not found for object with handle {obj_handle}")

                # Get object path
                object_path =object_template.handle
```
This method will be faster and will return the actual matching template directly.

Reply:
@jturner65  could you do the suggestion here: https://github.com/facebookresearch/habitat-lab/pull/1806
Siddharth's PR is not passing precommit and he has issues instaling it. The link I pointed to has the same code + linting